packages/ui/LogoSettingModal/actions.ts

```diff
@@ -0,0 +1,24 @@
+"use server";
+
+import { getServerSession } from "next-auth";
+
+import { authOptions } from "@formbricks/lib/authOptions";
+import { canUserAccessProduct, verifyUserRoleAccess } from "@formbricks/lib/product/auth";
+import { getProduct, updateProduct } from "@formbricks/lib/product/service";
+import { AuthorizationError } from "@formbricks/types/errors";
+import { TProductUpdateInput } from "@formbricks/types/product";
+
+export async function updateProductAction(productId: string, inputProduct: TProductUpdateInput) {
+  const session = await getServerSession(authOptions);
+  if (!session) throw new AuthorizationError("Not authorized");
+
+  const isAuthorized = await canUserAccessProduct(session.user.id, productId);
+  if (!isAuthorized) throw new AuthorizationError("Not authorized");
+
+  const product = await getProduct(productId);
+
```

Comment:
Add a check to throw an error if the product does not exist! Then you can also get rid of the force checker in 
 `product!.teamId`